Fix \ bug introduced in PR#15354 by removing parent of sub.

pkofod opened the pull request:
As @timholy pointed out in #15354 I, quite embarrassingly, introduced a bug in A\v for A being a `Factorization`, and v being a `SubArray` with complex elements, by inappropriately using `parent`. I have proposed a one-line fix, which does not at all use `SubArray`s in a good way, but it will give the correct result. I guess this is related to the whole discussion on inefficient fallback methods. I am happy to take advice and criticism if there's a better way.

_edit:_ didn't run all tests locally, so didn't catch the problems visible at CIs. I'll try to fix it.
_edit^2:_ fixed by defining it in the most general ("Factorization","SubArray") case.

pkofod committed:
Fix \ bug introduced in PR#15354 by removing parent of sub.

tkelman commented:
The bug only applies for the complex methods immediately above, right? And isn't the bug in those issues still there? What array types other than `SubArray` have `parent` differ from the array itself? It seems like the proper fix would be removing the usage of `parent`, but what would that break that you needed it for in #15354?

pkofod commented:
Ah, sorry. I obviously need to remove parent again. It went something like this.

1. The two methods didn't work for `(Factorization, SubArray)` (but wasn't tested before), because `reinterpret` doesn't work for `SubArray`s.
2. I added `parent` because I (mistakenly) thought it would fix the problem. It didn't, but there were no tests (like the one added here) to show it.
3. I added the new method to make sure the new test works.

tkelman commented:
There's probably a more efficient way to do `F\copy(B)` but I guess for now it's okay.
